config/default.yaml

```diff
@@ -189,6 +189,38 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+- apiGroups:
+  - sources.eventing.knative.dev
+  resources:
+  - gcppubsubsources
```

Comment:
if gcppubsub is not in default, the rbac rules should not be included for the resource.

I am not sure how hard this is to pull out.

Does rbac break if the resource is unknown to k8s?

Reply:
> I am not sure how hard this is to pull out.

I can't find a way to pull this out, other than a negative patch. There probably is a way, but I don't know Kubebuilder well enough to find it.

> Does rbac break if the resource is unknown to k8s?

I don't think so. I inserted this yaml and it was accepted:

```yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  creationTimestamp: null
  name: test
rules:
- apiGroups:
  - sources.eventing.knative.dev
  resources:
  - something-that-doesnt-exist
  verbs:
  - get
  - list
  - watch
  - create
  - update
  - patch
  - delete
```